tests/linalg_test.py

```diff
@@ -0,0 +1,93 @@
+from xdsl.ir import MLContext
+from xdsl.dialects.builtin import Builtin
+from xdsl.dialects.scf import Scf
+from xdsl.dialects.std import Std
+from xdsl.dialects.memref import MemRef
+from xdsl.dialects.affine import Affine
+from xdsl.dialects.arith import Arith
+from xdsl.dialects.linalg import Linalg
+from xdsl.parser import Parser
+from xdsl.printer import Printer
+
+
+def parse_and_verify(test_prog: str):
+    ctx = MLContext()
+    builtin = Builtin(ctx)
+    std = Std(ctx)
+    affine = Affine(ctx)
+    arith = Arith(ctx)
+    scf = Scf(ctx)
+    memref = MemRef(ctx)
+    linalg = Linalg(ctx)
+
+    parser = Parser(ctx, test_prog)
+    module = parser.parse_op()
+    module.verify()
+
+    printer = Printer()
+    printer.print_op(module)
+
+
+def test_linalg_generic_matmul():
+    test_prog = """
+module() {
```

Comment:
It might make sense to do this in a filecheck test. 
The `main` branch should now have some cleaner examples that show potential usages.